wordpress-mobile/WordPress-Android
# [Dependency Updates] Remove Unused Dependencies for Libs Modules

Parent #17558
Partially Closes: #17555

This issue is about removing all unused dependencies for the the `libs` related modules.

I recommend during a commit-by-commit review. You will also find additional information on the individual changes by reading a commit's description.

-----

Mocks Module List:

1. [Remove unnecessary jackson core exclusion for wiremock dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/d47691f646a83a8e3f1d29cc373ff66b40028bc6)
2. [Move apache http client android version to android test section.](https://github.com/wordpress-mobile/WordPress-Android/commit/3adae1068897bf925392edc743385b51fca64bc8)
3. [Rename apache http client android to wiremock http client version.](https://github.com/wordpress-mobile/WordPress-Android/commit/d62282c83428a98a10c2ee284bae4a976abed78d)

Image Editor Module List:

1. [Replace androidx lifecycle livedata ktx with core dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/2559a72b99590141fab1e2b4649f2661b65f87ea)
2. [Replace androidx lifecycle viewmodel ktx with viewmodel dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/a38490e47139b337a05080a7ba960230df2fce65)
3. [Replace androidx navigation ui ktx with ui dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/986875f492dd87a57e3eda0f681735a0edfcedc7)
4. [Replace androidx navigation fragment ktx with ui dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/9390e3755363231bef0352176a2f60656a6e2a75)
5. [Remove unused org jetbrains kotlin kapt plugin.](https://github.com/wordpress-mobile/WordPress-Android/commit/62188b8bc42d673eb86cd9606ff23ee273f86152)

Editor Module List:

1. [Remove unused androidx gridlayout dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/03279da499a077aef50da6fc29a18d0aa4b8fc6c)
2. [Use androidx legacy version for both support core ui and v4.](https://github.com/wordpress-mobile/WordPress-Android/commit/167955553f4e2fe9158c7c10d2e1ab2b60efece1)
3. [Remove unused androidx legacy support v4 dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/5cd63a91c9339239662711cf6dc0a7344b219090)
4. [Remove unused androidx vectordrawable animated dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/f2a8b18409c39242227691a2223f0168b4f40786)
5. [Remove unused jsoup dependency.](https://github.com/wordpress-mobile/WordPress-Android/commit/47e77d627b3a0396051197233b2b2cae193690c5)

Refactor List:

1. [Group androidx legacy dependencies together.](https://github.com/wordpress-mobile/WordPress-Android/commit/ab37810777c6a6d6bd875ce96bc4f4ff9defc3ad)
2. [Use androidx test espresso version for espresso accessibility.](https://github.com/wordpress-mobile/WordPress-Android/commit/838378d48256acddd612a7d0446aec137e339379)
3. [Group androidx test espresso dependencies together.](https://github.com/wordpress-mobile/WordPress-Android/commit/03dd9d196a079acd0c848c4f015e513e13987438)

-----

@oguzkocer I added you alone as the reviewer of this change as this will help us progress with the dependency updates, in terms of making the list of dependencies that we need to update more straightforward for us. Please expect another such PR that focuses solely on the `WordPress` module. I should have this ready on Monday... 🤞

-----

To test:

1. There is nothing much to test here.
2. Verifying that all the CI checks are successful should be enough.
3. However, if you want to be thorough about reviewing this change, you could:
    1. See the dependency tree diff result and verify correctness.
    2. Quickly smoke test both, the WordPress and Jetpack apps, and see if they both work as expected.
    3. Focus on testing the `Image Editor` and `Editor` related functionalities as this is were the bulk of the changes applied.

-----

## Regression Notes

1. Potential unintended areas of impact

Some of the dependencies removed might still be required as they might be hidden transitive dependencies and thus either crashing the app or causing some kind of misbehaviour.

4. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section.

9. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
